Added Time Format conversion using lua script through filter in "Lua filter" (#1417)
fluent/fluent-bit-docs: pipeline/filters/lua.md

                new_record["pub_date"] = result:match("%S+")
            end
        end
    end
    return 1, timestamp, new_record
end
```

#### Configuration

Configuration to get a json key with datetime and convert it to another format.

{% tabs %}
{% tab title="fluent-bit.conf" %}
```ini
[INPUT]
    Name    dummy
    Dummy   {"event": "Restock", "pub_date": "Tue, 30 Jul 2024 18:01:06 +0000"}
    Tag     event_category_a

[INPUT]
    Name    dummy
    Dummy   {"event": "Soldout", "pub_date": "Mon, 29 Jul 2024 10:15:00 +0600"}
    Tag     event_category_b

[FILTER]
    Name                lua
    Match               *
    Script              custom_datetime_format.lua
    call                convert_to_utc

[Output]
    Name                stdout
    Match               *
```
{% endtab %}

{% tab title="fluent-bit.yaml" %}
```yaml
pipeline:
  inputs:
    - name: dummy
      dummy: '{"event": "Restock", "pub_date": "Tue, 30 Jul 2024 18:01:06 +0000"}'
      tag: event_category_a

    - name: dummy
      dummy: '{"event": "Soldout", "pub_date": "Mon, 29 Jul 2024 10:15:00 +0600"}'
      tag: event_category_b

  filters:
    - name: lua
      match: '*'

>       script: custom_datetime_format.lua

Reviewer: Maybe do inline here in YAML format?

Author: @patrick-stephens Just pushed the changes. Will you please have a look? Thanks.

Reviewer: Yeah I think it's nicer to see the LUA inline in YAML now - having to do separate scripts is a PITA for helm, etc. too

#### Output

In the output only the messages with response code 0 or greater than 399 are shown.

### Timeformat Conversion

> In this example, we want to convert specific type of datetime format of a field to `utc ISO 8601` format.

Reviewer 2: ```suggestion
The following example converts a field's specific type of datetime format to
`utc ISO 8601` format.
```

#### Lua script

Script `custom_datetime_format.lua`

```lua
function convert_to_utc(tag, timestamp, record)
    local date_time = record["pub_date"]
    local new_record = record
    if date_time then
        if string.find(date_time, ",") then
            local pattern = "(%a+, %d+ %a+ %d+ %d+:%d+:%d+) ([+-]%d%d%d%d)"
            local date_part, zone_part = date_time:match(pattern)

            if date_part and zone_part then
                local command = string.format("date -u -d '%s %s' +%%Y-%%m-%%dT%%H:%%M:%%SZ", date_part, zone_part)
                local handle = io.popen(command)
                local result = handle:read("*a")
                handle:close()
                new_record["pub_date"] = result:match("%S+")
            end
        end
    end
    return 1, timestamp, new_record
end
```

#### Configuration

> Configuration to get a json key with datetime and convert it to another format.

Reviewer 2: ```suggestion
Use this configuration to obtain a JSON key with `datetime`, and then convert it to
another format.
```

{% tabs %}
{% tab title="fluent-bit.conf" %}
```ini
[INPUT]

>     Name    dummy

Reviewer 2: Can we rename this from "dummy" to something less pejorative? Like "placeholder" or "test" or something? And if we can, we'd need to do a search and replace in this to make that change, as it's used in code examples and at least one text line that follows.

Author: @cnorris-cs That `Name dummy` is actually another pipeline step from this: [https://docs.fluentbit.io/manual/pipeline/inputs/dummy](https://docs.fluentbit.io/manual/pipeline/inputs/dummy)

Reviewer: Yeah this comes up often @cnorris-cs but the actual input plugin is called `dummy` so is a required name for the configuration.
